kernel/vmemitcodes.ml

```diff
@@ -74,19 +135,19 @@ let patch1 buff pos n =
     (Char.unsafe_chr n) (Char.unsafe_chr (n asr 8))  (Char.unsafe_chr (n asr 16))
     (Char.unsafe_chr (n asr 24))
 
-let patch_int buff reloc =
+let patch_int buff reloc positions =
   let buff = Bytes.of_string buff in
-  let iter (reloc, npos) = Array.iter (fun pos -> patch1 buff pos reloc) npos in
-  let () = CArray.iter iter reloc in
+  let iter pos =
+    let id = Bytes.get_int32_le buff pos in
+    let reloc = reloc.(Int32.to_int id) in
+    patch1 buff pos reloc
```

Comment:
Since you use `Bytes.get_int32_le` to read this relocation, you might just as well use `Bytes.set_int32_le` to update its value. In fact, just get rid of `patch1`, since this code is the only user.